crates/starknet-devnet-server/src/builder.rs

```diff
@@ -103,6 +110,52 @@ impl<TJsonRpcHandler: RpcHandler, THttpApiHandler: Clone + Send + Sync + 'static
                     .allow_methods(vec![Method::GET, Method::POST]),
             );
 
+        if config.log_request {
```

Comment:
Why are log_response and log_request checks not consecutive?

Reply:
To avoid logging a response that is catched by the TimeoutLayer and also to avoid logging a request that exceeds the DefaultBodyLimit